lib/instance_billing_flavor_check/utils.py

```diff
@@ -139,6 +141,42 @@ def get_rmt_ip_addr():
     logger.info('Could not determine update server IP address')
 
 
+def _get_proxies():
+    """
+    Get proxy info.
+
+    If any proxy has been set up,
+    this method returns the proxy info in a dictionary.
+
+    Otherwise, returns None
+    """
+    if os.environ.get('http_proxy') or os.environ.get('https_proxy'):
+        logger.info('Using proxy settings present on the environment.')
+        # by default/if None requests relies on the environment variables
+        # HTTP_PROXY and HTTPS_PROXY
+        return None
+
+    proxies = {}
+    proxy_config = []
+    try:
+        with open(PROXY_CONFIG_PATH, 'r') as proxy_file:
+            proxy_config = proxy_file.readlines()
+    except FileNotFoundError:
+        logging.info('No proxy config file found on %s', PROXY_CONFIG_PATH)
+
+    for entry in proxy_config:
+        if 'PROXY_ENABLED' in entry and 'no' in entry:
+            return None
+        if 'HTTP_PROXY' in entry:
+            proxies['http_proxy'] = entry.split('"')[1]
+        if 'HTTPS_PROXY' in entry:
+            proxies['https_proxy'] = entry.split('"')[1]
+        if 'NO_PROXY' in entry:
+            proxies['no_proxy'] = entry.split('"')[1]
+
+    return proxies or None
```

Comment:
The empty dict will evaluate to `False` anywhere we have to check if there is a proxy as such the return of `None` is superfluous.